src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -300,25 +381,31 @@ func (r *Cluster) ValidateDelete() error {
 	return nil
 }
 
+// nolint:funlen,gocyclo // this function needs rewriting once stabilized
```

Comment:
oh gosh, this is the worst part of the codebase right now I think, I am refactoring it once this merges :D It was such a pain to handle even in my previous PRs

Reply:
Agree.